File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash = { nums[0]: 1}
        minimum = nums[0]

        for num in nums[1:]:
            if num not in hash:
                hash[num] = 1
            else:
                hash[num] += 1

            if hash[num] > hash[minimum]:
                minimum = num
            if hash[minimum] > len(nums) // 2:
                return minimum
        return nums[0]

File: Data Structures & Algorithms/first-missing-positive/submission-0.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        boolean[] seen = new boolean[nums.length + 1];

        for (int num: nums) {
            if (num > 0 && num <= nums.length) {
                seen[num] = true;
            }
        }

        for (int i = 1; i < seen.length; i++) {
            if (seen[i] == false) return i;
        }

        return nums.length + 1;
    }
}